File: segundo/1er cuatri/PDOO/irrgarten_entrega1/irrgarten_ruby/Irrgarten/monster.rb
# frozen_string_literal: true
require_relative 'dice'

class Monster
  @@INITIAL_HEALTH = 5

  def initialize(name, intelligence, strength)
    @name = name
    @intelligence = intelligence
    @strength = strength
    @health = @@INITIAL_HEALTH
    @row = -1
    @col = -1
  end

  def dead
    true if @health == 0
  end

  def attack
    Dice.intensity(@strength)
  end

  def defend(received_attack)
    is_dead = dead

    unless is_dead
      defensive_energy = Dice.intensity(@intelligence)

      if defensive_energy < received_attack
        got_wounded
        is_dead = dead
      end
    end

    is_dead
  end

  def to_s
    "[Nombre: #{@name}, intelligence: #{@intelligence}, strength: #{@strength}, health: #{@health}, row: #{@row}, col: #{@col} ]\n"
  end

  def inspect
    to_s
  end
  
  def got_wounded
    @health -= 1
  end

  def set_pos(row, col)
    @row = row
    @col = col
  end
end

File: segundo/1er cuatri/PDOO/irrgarten_entrega1/Irrgarten_java/src/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alvarogz
 */
public class Monster {
    private static final int INITIAL_HEALTH = 5;
    
    private String name;
    private float intelligence;
    private float strength;
    private float health;
    private int row;
    private int col;
    
    public Monster(String name, float intelligence, float strength) {
        this.name = name;
        this.intelligence = intelligence;
        this.strength = strength;
        this.health = INITIAL_HEALTH;
    }
    
    public boolean dead() {
        return this.health == 0;
    }
    
    public float attack() {
        return Dice.intensity(this.strength);
    }
    
    public boolean defend(float receivedAttack) {
        boolean isDead = dead();
        
        if (!isDead){
            float defensiveEnergy = Dice.intensity(intelligence);
            
            if (defensiveEnergy < receivedAttack){
                gotWounded();
                isDead = dead();
            }
        }
        
        return isDead;
    }
    
    public void setPos(int row, int col) {
        this.row = row;
        this.col = col;
    }
    
    @Override
    public String toString() {
        return "[Nombre: " + this.name + " , inteligencia: " + this.intelligence + ", fuerza: " + this.strength + ", vida: " + this.health + ", fila: " + this.row + ", columna: " + this.col + "]\n";
    }
    
    private void gotWounded() {
        this.health--;
    }   
 }
